### PR TITLE
feat: azure devops support for labels

### PR DESCRIPTION
**Pull request by JamieMagee**
Still need to write tests, but I've tested it and the functionality works.

### CODE REVIEW

File: lib/platform/azure/index.js [hunk 1]
```diff
@@ -522,8 +533,10 @@ async function addReviewers(prNo, reviewers) {
 }
 
 // istanbul ignore next
-function deleteLabel() {
-  throw new Error('deleteLabel not implemented');
+async function deleteLabel(prNumber, label) {
+  logger.debug(`Deleting label ${label} from #${issueNo}`);
+  const azureApiGit = await azureApi.gitApi();
+  await azureApiGit.deletePullRequestLabels(config.repoId, prNumber, label);
```
Comment (rarkins) [inline lib/platform/azure/index.js]: Make sure it doesn't throw an error if the label isn't found

**Review by rarkins — CHANGES_REQUESTED** [review]
Don't forget to add a `labels` field to the `getPr()` result too

File: lib/platform/azure/index.js (hunk 1 above)
Comment (JamieMagee) [reply lib/platform/azure/index.js]: It catches the error internally. See [here](https://github.com/Microsoft/azure-devops-node-api/blob/master/api/GitApi.ts#L3284-L3330)

**Comment by JamieMagee** [comment]
I'm having a lot of difficulty running tests locally. I get a lot of different snapshots. Have you seen anything like this before?

```sh
Snapshot Summary
 › 79 snapshots failed from 4 test suites. Inspect your code changes or run `yarn run jest -u` to update them.

Test Suites: 6 failed, 178 passed, 184 total
Tests:       99 failed, 1748 passed, 1847 total
Snapshots:   79 failed, 637 passed, 716 total
Time:        28.949s
Ran all test suites.
```

**Comment by rarkins** [comment]
@JamieMagee I've only seen before a small number of test failures, e.g. if someone doesn't have Python installed then there's at least one test that fails.

What do the failures say, and what's your platform/setup?

**Comment by JamieMagee** [comment]
Here's my environment
```sh
$ uname -a
Linux jamagee-laptop 5.0.2-arch1-1-ARCH #1 SMP PREEMPT Thu Mar 14 18:47:49 UTC 2019 x86_64 GNU/Linux
$ node --version
v11.12.0
```
And here's an example of the failures I'm getting

```sh
 FAIL  test/config/validation.spec.js
  ● config/validation › validateConfig(config) › returns nested errors

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "config/validation validateConfig(config) returns nested errors 1".

    - Snapshot
    + Received

    @@ -3,12 +3,12 @@
          "depName": "Configuration Error",
          "message": "Invalid configuration option: foo",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Invalid configuration option: lockFileMaintenance.bar",
    +     "message": "Unsafe regExp for packageRules[0].excludePackagePatterns: `(x+x+)+y`",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Unsafe regExp for packageRules[0].excludePackagePatterns: `(x+x+)+y`",
    +     "message": "Invalid configuration option: lockFileMaintenance.bar",
        },
      ]

      31 |       expect(warnings).toHaveLength(0);
      32 |       expect(errors).toHaveLength(3);
    > 33 |       expect(errors).toMatchSnapshot();
         |                      ^
      34 |     });
      35 |     it('included unsupported manager', async () => {
      36 |       const config = {

      at Object.toMatchSnapshot (test/config/validation.spec.js:33:22)

  ● config/validation › validateConfig(config) › errors for all types

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "config/validation validateConfig(config) errors for all types 1".

    - Snapshot
    + Received

      Array [
        Object {
          "depName": "Configuration Error",
    -     "message": "extends: Invalid timezone: Europe/Brussel",
    +     "message": "Configuration option `enabled` should be boolean. Found: 1 (number)",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Configuration option `enabled` should be boolean. Found: 1 (number)",
    +     "message": "Invalid schedule: `Schedule "every 15 mins every weekday" should not specify minutes`",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Configuration option `labels` should be a list (Array)",
    +     "message": "timezone: Invalid timezone: Asia",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Configuration option `lockFileMaintenance` should be a json object",
    +     "message": "Configuration option `labels` should be a list (Array)",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Configuration option `packageRules[3].packagePatterns` should be a list (Array)",
    +     "message": "Configuration option `semanticCommitType` should be a string",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Configuration option `semanticCommitType` should be a string",
    +     "message": "Configuration option `lockFileMaintenance` should be a json object",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Invalid configuration option: packageRules[1].foo",
    +     "message": "extends: Invalid timezone: Europe/Brussel",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Invalid regExp for packageRules[3].excludePackagePatterns: `abc ([a-z]+) ([a-z]+))`",
    +     "message": "Invalid configuration option: packageRules[1].foo",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "Invalid schedule: `Schedule "every 15 mins every weekday" should not specify minutes`",
    +     "message": "Configuration option `packageRules[3].packagePatterns` should be a list (Array)",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "packageRules must contain JSON objects",
    +     "message": "Invalid regExp for packageRules[3].excludePackagePatterns: `abc ([a-z]+) ([a-z]+))`",
        },
        Object {
          "depName": "Configuration Error",
          "message": "packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all.",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "timezone: Invalid timezone: Asia",
    +     "message": "packageRules must contain JSON objects",
        },
      ]

      93 |       );
      94 |       expect(warnings).toHaveLength(0);
    > 95 |       expect(errors).toMatchSnapshot();
         |                      ^
      96 |       expect(errors).toHaveLength(12);
      97 |     });
      98 |     it('selectors outside packageRules array trigger errors', async () => {

      at Object.toMatchSnapshot (test/config/validation.spec.js:95:22)

  ● config/validation › validateConfig(config) › selectors outside packageRules array trigger errors

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "config/validation validateConfig(config) selectors outside packageRules array trigger errors 1".

    - Snapshot
    + Received

      Array [
        Object {
          "depName": "Configuration Error",
    -     "message": "docker.minor.packageNames: packageNames should be inside a `packageRule` only",
    +     "message": "packageNames: packageNames should be inside a `packageRule` only",
        },
        Object {
          "depName": "Configuration Error",
    -     "message": "packageNames: packageNames should be inside a `packageRule` only",
    +     "message": "docker.minor.packageNames: packageNames should be inside a `packageRule` only",
        },
      ]

      116 |       );
      117 |       expect(warnings).toHaveLength(0);
    > 118 |       expect(errors).toMatchSnapshot();
          |                      ^
      119 |       expect(errors).toHaveLength(2);
      120 |     });
      121 |     it('ignore packageRule nesting validation for presets', async () => {

      at Object.toMatchSnapshot (test/config/validation.spec.js:118:22)
```

**Comment by rarkins** [comment]
Try Node.js 10.x - I noticed problems with someone else's Node.js 11.x previously

**Comment by JamieMagee** [comment]
Using Node.js 10.x seems to have worked. However, I noticed that all the errors with Node.js 11.x are due to the order of the objects being different. If you look at the error messages I posted above, all the error messages are still there but in the wrong order.

**Pull request merged**